portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/site/PortalSiteHelperImpl.java

```diff
@@ -1418,5 +1454,19 @@ public Site getSite(String siteId) throws IdUnusedException
 			throw e;
 		}
 	}
+	
+	public String getRole(String siteId, String userId) {
+		String role = "";
+		try {
+			if(null!=getAuthzGroupService()) {
+				AuthzGroup realm = getAuthzGroupService().getAuthzGroup("/site/" + siteId);
+				role = getAuthzGroupService().getUserRole(userId, realm.getId());
+			}
+		}
+		catch (GroupNotDefinedException e) {
+			log.error("GroupNotDefinedException:", e);
```

Comment:
We'll know this is a GroupNotDefinedException from the stack trace. Why not put a more useful piece of text here?

Reply:
Added new text.